keps/sig-auth/3299-kms-v2-improvements/README.md

```diff
@@ -0,0 +1,579 @@
+What signals should users be paying attention to when the feature is young
+that might indicate a serious problem?
+-->
+
+This section is incomplete and will be updated before the beta milestone.
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+<!--
+Describe manual testing that was done and the outcomes.
+Longer term, we may want to require automated upgrade/rollback tests, but we
+are missing a bunch of machinery and tooling and can't do that now.
+-->
+
+This section is incomplete and will be updated before the beta milestone.
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+<!--
+Even if applying deprecation policies, they may still surprise some users.
+-->
+
+N/A
+
+### Monitoring Requirements
+
+###### How can someone using this feature know that it is working for their instance?
+
+- [x] Other (treat as last resort)
+  - Details:
+    - Logs in kube-apiserver, kms-plugin and KMS will be logged with the corresponding `observed_key_id`, `annotations`, and `UID`.
+    - count of encryption/decryption requests by resource and version
+
+###### What are the reasonable SLOs (Service Level Objectives) for the enhancement?
+
+There should be no impact on the SLO with this change.
+
+###### What are the SLIs (Service Level Indicators) an operator can use to determine the health of the service?
+
+- [x] Other (treat as last resort)
+  - Details:
+    - Logs in kube-apiserver, kms-plugin and KMS will be logged with the corresponding `observed_key_id`, `annotations`, and `UID`.
+    - Metrics for latency of encryption/decryption requests.
+
+### Dependencies
+
+###### Does this feature depend on any specific services running in the cluster?
+
+No.
+
+### Scalability
+
+###### Will enabling / using this feature result in any new API calls?
+
+Yes, the new KMS v2 gRPC API.
+
+###### Will enabling / using this feature result in introducing new API types?
+
+Yes, the new KMS v2 gRPC types.
+
+###### Will enabling / using this feature result in any new calls to the cloud provider?
+
+No.
+
+###### Will enabling / using this feature result in increasing size or count of the existing API objects?
+
+No, the v2 API is new.
+
+###### Will enabling / using this feature result in increasing time taken by any operations covered by existing SLIs/SLOs?
+
+No.
+
+###### Will enabling / using this feature result in non-negligible increase of resource usage (CPU, RAM, disk, IO, ...) in any components?
+
+No.
+
+### Troubleshooting
+
+###### How does this feature react if the API server and/or etcd is unavailable?
+
+- This feature is part of API server. The feature is unavailable if API server is unavailable.
+- ETCD data encryption with external kms-plugin is unavailable
+- If the API server is unavailable, clients will be unable to create/get data that's stored in etcd. There will be no requests from the API server to the kms-plugin.
```

Comment:
If the KMS configuration is invalid, the healthz and readyz behavior will be same as today. For the hot reload we're proposing, any changes to KMS configuration that is not valid will be ignored when the API server is still running. However if the API server restarts and the KMS configuration is invalid, the behavior exactly same as today.

Reply:
Mind indicating the value of readyz and healthz here so, "what happens when there is an outage" is clear when someone has to look at this  to figure out what the impact to his  cluster is?

Reply:
Added the `/livez` and `/readyz` details with different scenarios it could fail with. Please let us know if there is more information we could add to the PRR.